File: PS/Programmers/스킬트리.cc
#define _CRT_SECURE_NO_WARNINGS
#include <iostream>
#include <string>
#include <vector>
#include <map>
#include <queue>
#include <algorithm>
#include <stack>
#include <sstream>
#include <bitset>
#include <cstring>
using namespace::std;


int solution(string skill, vector<string> skill_trees) {
    int answer = 0;

    int s = skill.size();
    for (int i = 0; i < skill_trees.size(); i++)
    {
        int id = 0;
        bool check = true;
        for (int j = 0; j < skill_trees[i].size(); j++)
        {
            for (int a = 0; a < s; a++)
            {
                if (skill_trees[i][j] == skill[a])
                {
                    if (id == a)
                    {
                        id++;
                    }
                    else {
                        check = false;
                        break;
                    }
                }
            }
        }
        if (check)
            answer++;
    }
    return answer;
}

File: PS/Programmers/스킬트리.py
def solution(skill, skill_trees):
    answer = 0
    for i in skill_trees:
        id = 0
        check = True
        for j in range(len(i)):
            for a in range(len(skill)):
                if i[j] == skill[a]:
                    if id == a: id+=1
                    else: 
                        check = False
                        break
        if check: answer += 1
    return answer